client/app.jsx

```diff
@@ -116,5 +195,12 @@ class Carousel extends React.Component {
   }
 }
 
+$.ajax({
```

Comment:
This GET is requesting for the product information - do you also have a GET for the urls? If so, I think you can put the ajax request in a componentDidMount since you want the data that contains the urls from a remote endpoint.